Part #19737 - Replace deprecated FormField component with TextField, Label, HelpText

Jainex17 opened the pull request:
## Explanation

This pull request addresses the Issue of replacing the deprecated `FormField` component with the modernized `Label`, `TextField` and `HelpText` components from our component library. Additionally, it includes updates to the constants previously used in the codebase, changing `DISPLAY` to `Display`, `FLEX_DIRECTION` to `FlexDirection`, and `BLOCK_SIZES` to `BlockSize`.

* Part #19737 

## Changes made
* Replaced instances of the deprecated `FormField` component with `TextField`, `Label` and `HelpText` components.
* Updated constants: `DISPLAY` to `Display`, `FLEX_DIRECTION` to `FlexDirection`, and `BLOCK_SIZES` to `BlockSize`.

## Screenshots/Screencaps
### Before
Some of the icons and text were unaligned and inconsistent. The input type is also text when it is a number input 

https://github.com/MetaMask/metamask-extension/assets/8112138/aef8a0d1-cbed-4a92-9431-d985020f1dad

Storybook also didn't work as expected you can't edit the input

https://github.com/MetaMask/metamask-extension/assets/8112138/289c3e96-48ac-484e-bad2-4ba91558ae39

### After
Icons are now consistent in the label and tooltip, the alignment of the max button is centered the label has also been updated to align with the design system


https://github.com/MetaMask/metamask-extension/assets/8112138/8a500fdb-e394-4463-82a6-98ed521db3d6


Storybook now works as expected being able to use the input to enter numbers

https://github.com/MetaMask/metamask-extension/assets/8112138/c120fd5c-6ce2-46ca-a515-ac11331dce4d


## Manual Testing Steps
Storybook
- Pull this branch
- Serach `CustomSpendingCap` in stroybook
- Check against `develop` storybook https://metamask.github.io/metamask-storybook/?path=/story/components-app-customspendingcap--custom-spending-cap-story&args=dappProposedValue:234234234

Check custom spending cap in app
- Go to test dapp https://metamask.github.io/test-dapp/
- Click `CREATE TOKEN` you will need test eth
- Wait for transaction
- Click `APPROVE TOKENS` 
- Use Custom Spending Cap component 

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

Jainex17 commented:
Hello @georgewrmarshall 

I've submitted a pull request with several changes and improvements to the codebase. However, there are currently three failing tests in the PR:

* ci/circleci: test-e2e-chrome — Failed on CircleCI
* ci/circleci: test-e2e-firefox — Failed on CircleCI
* ci/circleci: test-unit-jest-main

If you have the time would you be able to lend a hand in resolving these failing tests? Your expertise would be greatly appreciated. However, if the issues are complex or if you are occupied with other tasks, I'm completely open to closing this PR and working on a different issue. Please let me know your preferred course of action.

Thank you

georgewrmarshall commented:
Hey @Jainex17, any chance you could rebase and try again? I will look into the e2e test failures once that's done

Jainex17 commented:
Hey @georgewrmarshall, I've pulled the latest changes and addressed the issues from scratch. I've also made some modifications and drop code in resolve option both js and snap files. However that tests are still failing sorry for that.🥲

# screenshots
## Before
![](https://github.com/MetaMask/metamask-extension/assets/81921291/b837b814-6472-48ba-bb70-61fcc919b21e)

## After
![](https://github.com/MetaMask/metamask-extension/assets/81921291/4ea4b9d1-91e5-42ca-83b7-bc4939a039a7)

georgewrmarshall commented:
Hey @Jainex17, Would it be possible to update this PR?

Jainex17 commented:
Hey @georgewrmarshall, I resolved the conflict. Could you please review it?

Jainex17 commented:
> Hey @Jainex17, Would it be possible to update this PR?

Hey @georgewrmarshall,  I have made updates to some of the code to fix the Jest errors, and while some of them have been resolved successfully, a few still remain. I have attempted to address those remaining issues but, unfortunately, have not been able to resolve them.

If you have the time, could you please take a look at the remaining error? I have attached a screenshot after making some changes.

Thank you for your assistance.

![Screenshot 2023-07-26 133939](https://github.com/MetaMask/metamask-extension/assets/81921291/cb38744b-9207-4960-84ec-7634dd4c057e)

georgewrmarshall commented on ui/components/app/custom-spending-cap/custom-spending-cap.js:
```diff
@@ -231,16 +238,26 @@ export default function CustomSpendingCap({
+              >
+                <Label
+                  htmlFor="custom-spending-cap"
+                  fontWeight={FontWeight.Bold}
```
I think we can remove the fontWeight here
```suggestion
```

Jainex17 replied:
> I think we can remove the fontWeight here

Removing fontWeight may cause a slight difference from the original font.

I have attached both the `before` and `after` screenshots for your review. If everything appears satisfactory, we can proceed with the changes.

<table>
<tr>
<td>original</td>
<td>before remove</td>
<td>After remove</td>
</tr>
<tr>

<td><img src="https://github.com/MetaMask/metamask-extension/assets/81921291/64431bc7-8723-4289-8101-4a2c2677c07e" /> </td>

<td> <img src="https://github.com/MetaMask/metamask-extension/assets/81921291/f1700574-855c-4cec-89f0-14cc457bf68a"/> </td>

<td> <img src="https://github.com/MetaMask/metamask-extension/assets/81921291/b75aabd3-1932-47fd-96eb-4833a033206d" /> </td>

</tr>
</table>

georgewrmarshall replied:
Great spotting @Jainex17 the design team has update the `Label` to use medium font weight so we can align them here

Jainex17 commented:
Hello I've addressed the lint error, but some Jest tests are still failing. Could you please take a look?

georgewrmarshall commented on test/e2e/tests/custom-token-add-approve.spec.js:
```diff
@@ -376,7 +376,7 @@ describe('Create token, approve token and approve token without gas', function (
         // set max spending cap
         await driver.clickElement({
-          css: '.custom-spending-cap__max',
+          css: '[data-testid="custom-spending-cap-max-button"]',
```
Updating to use data-testid

georgewrmarshall commented on ui/components/app/app-components.scss:
```diff
@@ -20,7 +20,6 @@
 @import 'connected-accounts-permissions/index';
 @import 'connected-sites-list/index';
 @import 'create-new-vault/create-new-vault.scss';
-@import 'custom-spending-cap/index';
```
Removing unused styles

georgewrmarshall committed:
Replacing deprecated components for component library components

georgewrmarshall committed:
Merge branch 'develop' into issue/Replacing-FormField

georgewrmarshall committed:
Merge branch 'develop' into issue/Replacing-FormField

georgewrmarshall committed:
Update

georgewrmarshall committed:
Lint fix

garrettbear reviewed (approved):
LGTM

georgewrmarshall reviewed (approved):
LGTM
